packages/react-i18n/src/index.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __, _n, _nx, _x } from '@wordpress/i18n';
+
+export const I18nContext = React.createContext< undefined | string >( undefined );
+
+interface I18nTranslationFunctions {
+	__: typeof __;
+	_n: typeof _n;
+	_nx: typeof _nx;
+	_x: typeof _x;
+}
+
+/**
+ * React hook providing i18n translate functions
+ *
+ * @example
+ *
+ * import { useI18n } from '@automattic/react-i18n';
+ * function MyComponent() {
+ *   const { __ } = useI18n();
+ *   return <div>{ __( 'Translate me.' ) }</div>;
```

Comment:
Maybe add translation domain to the example?

```suggestion
 *   return <div>{ __( 'Translate me.', 'textdomain' ) }</div>;
```